modules/local/vembrane/create-vembrane-fields/main.nf

```diff
@@ -14,46 +20,61 @@ process VEMBRANE_CREATE_FIELDS {
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
+    def args   = task.ext.args   ? (task.ext.args.endsWith(",") ? task.ext.args : task.ext.args + ",") : '' 
     def prefix = task.ext.prefix ?: "${meta.id}"
 
     // check if vcf file is gzipped
     def vcf_list = vcf.collect { it.toString() }
     def vcf_zip  = vcf_list[0].endsWith('.gz')
     def command1 = vcf_zip ? 'zcat' : 'cat'
 
-    if ( extraction_fields == 'all' )
-        """
-        # get CSQ field names and convert to vembrane format
-        csqfields=\$($command1 $vcf | awk -F'Format: ' '/CSQ/ && /^#/{ \\
-            gsub(/">/, "", \$2); \\
-            split(\$2, fields, "\\\\|"); \\
-            output = ""; \\
-            for (i = 1; i <= length(fields); i++) { \\
-                if (output != "") output = output ", "; \\
-                output = output "CSQ[\\"" fields[i] "\\"]" \\
-            }; \\
-            print output \\
-        }') \\
-
-        # get CSQ field names for header
-        csqheader=\$($command1 $vcf | awk -F'Format: ' '/CSQ/ && /^#/{
+    // check if strings have comma in the end
+    def format_field_withend = format_fields ? (format_fields.endsWith(",")  ? format_fields : format_fields + ",") : ''
+    def info_field_withend   = info_fields   ? (info_fields.endsWith(",")    ? info_fields   : info_fields + ",")   : ''
+
+    """
+    #get or set CSQ annotation fields
+    if [[ '$annotation_fields' == 'all' ]]; then
```

Comment:
This code works in terms of nextflow variables. Maybe it can be integrated as groovy if/else statements before the code lines (i.e., the three quotes)?